repository/repository.go

```diff
@@ -457,3 +464,75 @@ To find more information on specific steps, please see documentation [here](http
 - [ ] QA: Final validation of above PR and tracked through the linked ticket
 - [ ] PJM: Close the milestone in GitHub.
 `
+
+type Author struct {
+	Name  string `json:"name"`
+	Email string `json:"email"`
+	Date  string `json:"date"`
+}
+
+type Commit struct {
+	Author    Author `json:"author"`
+	Committer Author `json:"committer"`
+	Message   string `json:"message"`
+	URL       string `json:"url"`
+}
+
+type CommitResponse struct {
+	SHA         string `json:"sha"`
+	NodeID      string `json:"node_id"`
+	Commit      Commit `json:"commit"`
+	URL         string `json:"url"`
+	HTMLURL     string `json:"html_url"`
+	CommentsURL string `json:"comments_url"`
+}
+
+func CommitInfo(owner, repo, commitHash string, httpClient *http.Client) (*Commit, error) {
+	apiUrl := fmt.Sprintf(baseURLGHApi+"/repos/%s/%s/commits/%s", owner, repo, commitHash)
+
+	response, err := httpClient.Get(apiUrl)
+	if err != nil {
+		logrus.Debug("Error making request:", err)
+		return nil, err
+	}
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusOK {
+		return nil, errors.New("failed to fetch commit information. status code: " + strconv.Itoa(response.StatusCode))
+	}
+	data, err := ioutil.ReadAll(response.Body)
```

Comment:
I don't think this is necessary or efficient. Let's update the JSON code below to use a JSON Decoder and remove this block of code.